ccips/ccip-022/ccip-022-citycoins-treasury-redemption-nyc.md

```diff
@@ -0,0 +1,66 @@
+# CCIP-022
+
+## Preamble
+
+| CCIP Number   | 022                                                  |
+| ------------- | ---------------------------------------------------- |
+| Title         | CityCoins Treasury Redemption NYC                    |
+| Author(s)     | Raphael R. Sierra rapha@fontainebleau-management.com |
+|               | Jason Schrader jason@joinfreehold.com                |
+| Consideration | Governance                                           |
+| Type          | Standard                                             |
+| Status        | Draft                                                |
+| Created       | 2024-04-23                                           |
+| License       | BSD-2-Clause                                         |
+| Supplements   | CCIP-020                                             |
+
+## Introduction
+
+With the execution of CCIP-020 Graceful Protocol Shutdown[^1] the functions for mining and stacking CityCoins are disabled.
+
+Following that change this CCIP proposes a redemption mechanism for NYC token holders to be fairly compensated by redeeming their CityCoins for a portion of the STX held in the city treasury.
+
+## Specification
+
+### Redemption Extensions
+
+This CCIP will implement a new redemption extension for NYC.
+
+The redemption extensions will:
+
+- receive and hold funds from the respective city treasury contract
+- record the total supply of NYC at the time of initialization
+- calculate the ratio of NYC to STX based on the total supply and contract balance
+- allow users to burn NYC in exchange for their portion of STX and track the related totals
```

Comment:
```suggestion
- allow users to burn NYC in exchange for their portion of STX and track the related totals
- allow unclaimed STX to be stacked using delegated stacking
- allow unclaimable STX (due to burnt, but not redeemed NYC tokens or rounding errors) to be transferred to a different extension
```

Reply:
These two bullet points are not covered in the current contract code, and after a lot of the back and forth discussion it still seems best to keep this simple. If you have NYC you can redeem for STX at any time. There is no way we can estimate the locked/lost amounts and putting a deadline is unnecessary pressure for holders.